pkg/sbproxy/reconcile/reconcile_visibilities.go

```diff
@@ -301,9 +367,9 @@ func (r *ReconciliationTask) getVisibilityKey(visibility *platform.ServiceVisibi
 
 	const idSeparator = "|"
 	if visibility.Public {
-		return strings.Join([]string{"public", "", visibility.CatalogPlanID}, idSeparator)
+		return strings.Join([]string{"public", "", visibility.BrokerID, visibility.CatalogPlanID}, idSeparator)
```

Comment:
this won't work. the platform will return you visibilities with broker ID different from the SM broker id. You either need to override the brokerID returned from the client (cf-proxy/k8s-proxy) with the SM brokerID, or you need some other type of key matching

Reply:
This is the broker ID from the Service Manager, unless I made some mistake along the way.